components/button/button.tsx

```diff
@@ -180,6 +180,8 @@ const InternalCompoundedButton = React.forwardRef<
   const loadingOrDelay = useMemo<LoadingConfigType>(() => getLoadingConfig(loading), [loading]);
 
   const [innerLoading, setLoading] = useState<boolean>(loadingOrDelay.loading);
+  // Fix https://github.com/ant-design/ant-design/issues/51325
+  const innerLoadingRef = useRef<boolean>(loadingOrDelay.loading);
```

Comment:
loading state should be unique. If use `ref` for state control. Origin `innerLoading` can be replaced with useForceUpdate instead.